test/SlowTests/Server/Documents/DataArchival/DataArchivalTests.cs

```diff
@@ -71,296 +64,68 @@ public async Task CanSetupDataArchival()
-        //                 Assert.Equal(docsPerCollection  * 2, usersArchivedCountersTable.NumberOfEntries);
-        //                 Assert.Equal(docsPerCollection * 2 ,companiesArchivedCountersTable.NumberOfEntries);
-        //                 
-        //                 var archivedUsersCounters = database.DocumentsStorage.CountersStorage.GetCountersFrom(ctx, "UsersArchived", 0, 0, int.MaxValue).ToList();
-        //                 var archivedCompanyCounters = database.DocumentsStorage.CountersStorage.GetCountersFrom(ctx, "CompaniesArchived", 0, 0, int.MaxValue).ToList();
-        //                 Assert.Equal(docsPerCollection * 2,archivedUsersCounters.Count);
-        //                 Assert.Equal(docsPerCollection * 2, archivedCompanyCounters.Count);
-        //             }   
-        //             
-        //         }
-        //     }
-        // }
+        [Fact]
+        public async Task CanArchiveEntity_WillNotBeRelevantInAutoMapIndex()
+        {
+            using (var store = GetDocumentStore())
+            {
+                // Insert document with archive time before activating the archival
+                var company1 = new Company {Name = "Company Name"};
+                var retires = SystemTime.UtcNow.AddMinutes(5);
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(company1);
+                    var metadata = session.Advanced.GetMetadataFor(company1);
+                    metadata[Constants.Documents.Metadata.Archive] = retires.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                    await session.SaveChangesAsync();
+                }
+
+                // Activate the archival
+                await SetupDataArchival(store);
+
+                var database = await Databases.GetDocumentDatabaseInstanceFor(store);
+                database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+                var documentsArchiver = database.DataArchivist;
+                await documentsArchiver.ArchiveDocs();
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var company2 = await session.LoadAsync<Company>(company1.Id);
+                    var metadata = session.Advanced.GetMetadataFor(company2);
+                    Assert.DoesNotContain(Constants.Documents.Metadata.Archive, metadata.Keys);
+                    Assert.Contains(Constants.Documents.Metadata.Collection, metadata.Keys);
+                    Assert.Contains(Constants.Documents.Metadata.Archived, metadata.Keys);
+                    Assert.Equal(true, metadata[Constants.Documents.Metadata.Archived]);
+
+                    List<Company> companies = await session.Query<Company>().Where(x => x.Name == "Company Name").ToListAsync();
+                    Assert.Equal(0, companies.Count);
```

Comment:
Where do you assert that results were returned _before_ the doc archival?

Reply:
Added that missing part